http.go

```diff
@@ -233,20 +229,8 @@ func (request *HTTPRequest) DoHTTPAction() (payloadBytes []byte, responseStatusC
 		}
 	}
 	resp, doErr := request.Client.Do(req)
-	if doErr != nil {
-		//If we tried the secure client and it got an error, try the insecure one
-		if (request.TLSClientPair != nil) &&
-			(request.TLSClientPair.SecureClient != nil) &&
-			(request.TLSClientPair.SecureClient != request.TLSClientPair.InsecureClient) {
-			log.Printf("INFO: Cert-validating HTTP client request failed, trying non-validating client.")
-			request.Client = request.TLSClientPair.InsecureClient
-			resp, doErr = request.Client.Do(req)
-		}
-	}
```

Comment:
SMD also uses hms-go-http-lib's DoHTTPAction, and maybe other hms projects do too.
Do you know if SMD still works with this?